lib/internal/process/esm_loader.js

```diff
@@ -56,8 +57,24 @@ async function initializeLoader() {
 
   const { getOptionValue } = require('internal/options');
   const customLoaders = getOptionValue('--experimental-loader');
+  const preloadModules = getOptionValue('--experimental-import');
+  const keyedExportsList = await loadInternalModules(customLoaders);
 
-  if (customLoaders.length === 0) return;
+  // Hooks must then be added to external/public loader
+  // (so they're triggered in userland)
+  await esmLoader.addCustomLoaders(keyedExportsList);
+
+  // Preload after loaders are added so they can be used
+  if (preloadModules?.length) {
+    await loadInternalModules(preloadModules);
+  }
```

Comment:
Should preload modules be imported _before_ custom loaders so they're available in them, or should we leave that to ambient loaders?

Reply:
I would expect loaders to import whatever thay need, and imports to be evaluated after loaders, if for example I want to `--import` a typescript file